src/frontend/komponenter/Fagsak/Behandle/FastsettVedtak.tsx

```diff
@@ -102,9 +110,41 @@ const FastsettVedtak: React.FunctionComponent<IProps> = ({ fagsak }) => {
                             method: 'POST',
                             url: `/familie-ba-sak/api/fagsak/${fagsak.id}/nytt-vedtak`,
                         })
-                            .then((response: Ressurs<any>) => {
+                            .then((response: Ressurs<IVedtakForBehandling>) => {
                                 dispatch({ type: actions.SETT_SENDER_INN, payload: false });
                                 if (response.status === RessursStatus.SUKSESS) {
+                                    fagsakDispatch({
```

Comment:
Why do we want to do this? Everything should load fine when redirecting to the next page.

Reply:
We need to update the Fagsak context so that the vedtak page knows the updated vedtak information.
Just like you updated the Fagsak context from step 1 to step 2

Reply:
Right, but then I would suggest changing the api to return the whole fagsak so that we dont need to do so much in the frontend.

Reply:
This is also a viable solution. But it does not eliminate the complexity: we need to do even more in the backend. Just like what you did in hentRestFagsak(), we need to make a couple of queries to setup the fagsak.
If we use bi-directional mapping in the entity model, it could be much simpler. However the queries are still necessary, although hibernate can hide them from us.
I prefer frontend computation, because it can improve the system performance at least.

Reply:
I see the hibernate mapping, you can fix it if you want 👍 
I also awknowledge that it is nice to do things frontend, but the data is minimal and the function we need to add the vedtak is pretty messy and hard to understand.

Reply:
Also, the backend does not return a "Rest" object, it returns the hibernate object which we actually dont want to use for the frontend (my bad :( )

Reply:
Maybe we can fix the hibernate stuff after Christmas?
Well, you are right and I see that the understandability has become a issue🙂 I will use the backend solution then👍

Since you have merged quite a few stuff. I will remove this PR and create a new one based on yours.

Reply:
Cool, ok 👍